2021-Firmware-Bootcamp/Core/Src/main.c

```diff
@@ -0,0 +1,201 @@
+  *
+  ******************************************************************************
+  */
+/* USER CODE END Header */
+/* Includes ------------------------------------------------------------------*/
+#include "main.h"
+#include "spi.h"
+#include "tim.h"
+#include "usart.h"
+#include "gpio.h"
+
+/* Private includes ----------------------------------------------------------*/
+/* USER CODE BEGIN Includes */
+#include "stm32f0xx_hal.h"
+#include "stm32f0xx_hal_spi.h"
+#include <stdint.h>
+#include <stdio.h>
+
+/* USER CODE END Includes */
+
+/* Private typedef -----------------------------------------------------------*/
+/* USER CODE BEGIN PTD */
+
+/* USER CODE END PTD */
+
+/* Private define ------------------------------------------------------------*/
+/* USER CODE BEGIN PD */
+/* USER CODE END PD */
+
+/* Private macro -------------------------------------------------------------*/
+/* USER CODE BEGIN PM */
+
+/* USER CODE END PM */
+
+/* Private variables ---------------------------------------------------------*/
+
+/* USER CODE BEGIN PV */
+
+/* USER CODE END PV */
+
+/* Private function prototypes -----------------------------------------------*/
+void SystemClock_Config(void);
+/* USER CODE BEGIN PFP */
+
+/* USER CODE END PFP */
+
+/* Private user code ---------------------------------------------------------*/
+/* USER CODE BEGIN 0 */
+
+/* USER CODE END 0 */
+
+/**
+  * @brief  The application entry point.
+  * @retval int
+  */
+int main(void)
+{
+  /* USER CODE BEGIN 1 */
+
+  /* USER CODE END 1 */
+
+  /* MCU Configuration--------------------------------------------------------*/
+
+  /* Reset of all peripherals, Initializes the Flash interface and the Systick. */
+  HAL_Init();
+
+  /* USER CODE BEGIN Init */
+  uint8_t rx_buffer_split[2];
+  uint8_t tx_buffer[2];
+  rx_buffer_split[0] = 0;
+  rx_buffer_split[1] = 0;
+  tx_buffer[0] = 1;
+  tx_buffer[1] = 0b1001;
+  /* USER CODE END Init */
+
+  /* Configure the system clock */
+  SystemClock_Config();
+
+  /* USER CODE BEGIN SysInit */
+
+  /* USER CODE END SysInit */
+
+  /* Initialize all configured peripherals */
+  MX_GPIO_Init();
+  MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
+  /* USER CODE BEGIN 2 */
+  HAL_TIM_Base_Start(&htim1);
+  /* USER CODE END 2 */
+
+  /* Infinite loop */
+  /* USER CODE BEGIN WHILE */
+  while (1)
+  {
+    /* USER CODE END WHILE */
+	  if(HAL_SPI_Transmit(&hspi1, &tx_buffer[0], sizeof(tx_buffer[0]), 0) == HAL_ERROR) continue;
+	  if(HAL_SPI_TransmitReceive(&hspi1, &tx_buffer[1], &rx_buffer_split[0], sizeof(tx_buffer[1])+sizeof(rx_buffer_split[0]), 0) == HAL_ERROR) continue;
+	  if(HAL_SPI_Receive(&hspi1, &rx_buffer_split[1], 8, 0) == HAL_ERROR) continue;
+	  rx_buffer_split[0] = rx_buffer_split[0] & 0b00000011; //isolating B9 and B8
+	  uint16_t rx_buffer = (rx_buffer_split[0] << 8) + rx_buffer_split[1];
+	  if(rx_buffer > 65535 || rx_buffer < 0) continue;
+	  HAL_TIM_PWM_ConfigChannel(&htim1, rx_buffer, TIM_CHANNEL_1);

```

Comment:
I don't think this function is the correct function to manipulate the PWM duty cycle here. You are also missing a piece of math to convert the ADC data to PWM timer counts. 
ADC generates a number that vary from 0 to (2^10-1) since the resolution of this ADC is 10 bits. 
Your PWM timer has 65535 counts in one period. When 5% of 65535 is on high level, 95% of counts on low level, you have a 5% duty cycle and that maps to 0 from the ADC data. 
If the PWM has a 10% duty cycle, meaning 10% of 65535 counts on high, 90% on low. That maps to the max number from ADC.
Your math is supposed to handle this mapping from ADC value to PWM counts, and find the correct HAL function to change the Timer from high to low.